### PR TITLE
Sb es modal test

### LINKED ISSUE #15: 13. As a user, I want to view details of my purchases to better understand my purchase patterns
AC:

* From the shopping list, user can tap to navigate to a detailed view of the item
* Detail view includes the following information:
  * Name of the item
  * Last purchased date
  * Estimated number of days until the next purchase
  * Number of times the item has been purchased"
* When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view

### PR DESCRIPTION
**Pull request by espain16**
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Related Issue

closes #15 
## Description

From the shopping list, the user can tap to navigate to a detailed view of the item.
The detail view includes the following information:
- Name of the item
- Last purchased date
- Estimated number of days until the next purchase
- Number of times the item has been purchased"
When in the “item details” view, a back arrow is displayed in the header to take the user back to the “list” view.

## List To Complete Task
- [x] Create a component called ItemDetails.
- [x] Create a route for Item-Details.
- [x] Add link to each item to view the item's details.
- [x] Add a back arrow on the details page to take the user back the Shopping list.
- [x] Format the last purchased date to be human readable.

#### Previous Shopping List 
Screen Shot 2020-05-19 at 11 32 26 AM

#### Update To Shopping List UI 
Screen Shot 2020-05-19 at 11 28 34 AM

#### Item Detail View
Screen Shot 2020-05-21 at 10 44 23 AM

## Acceptance Criteria

* From the shopping list, user can tap to navigate to a detailed view of the item
* Detail view includes the following information:
  * Name of the item
  * Last purchased date
  * Estimated number of days until the next purchase
  * Number of times the item has been purchased"
* When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view

## Testing Steps / QA Criteria

Click [Here](https://deploy-preview-41--tcl-6-smart-shopping-list.netlify.app/) for the preview of our branch. 
- Click view details
- Click back to shopping or the close button to return to list view.

### CODE REVIEW

File: src/CSS/TestModal.css [hunk 1]
```diff
@@ -0,0 +1,235 @@
+/** Modal Styling **/
```
Comment (stevelikesmusic) [inline src/CSS/TestModal.css:1]: I would probably include a comment linking to the source of this CSS

File: src/components/ListItem.js [hunk 2]
```diff
@@ -55,26 +65,30 @@ const ListItem = ({ item, onDelete, token }) => {
...
+        >
+          Purchase
+          <span className="screen-reader-only">{item.name}.</span>
+        </button>
+        <button onClick={onDelete}>
+          <img className="trash" src={trash} alt="delete item" />
+        </button>
+        <button onClick={handleShow}>View Details</button>
+      </li>
+      <TestModal item={item} show={show} handleClose={handleClose} />
```
Comment (stevelikesmusic) [inline src/components/ListItem.js]: This is great! Nice way to show the item details. Another option using a different library might be [Drawer](https://material-ui.com/components/drawers/) from Material UI

File: src/components/TestModal.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import { Link } from 'react-router-dom';
+import arrowLeft from '../image/arrow-left.svg';
+import '../CSS/Icon.css';
+
+const TestModal = ({ show, handleClose, item }) => {
+  let lastPurchase = item ? new Date(item.last_purchased).toDateString() : '';
+
+  return (
+    <>
```
Comment (stevelikesmusic) [inline src/components/TestModal.js:16]: You shouldn't need the Fragment if you have a wrapping component like `Modal`.

File: src/components/TestModal.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      <Modal show={show} onHide={handleClose}>
+        <Modal.Header>
```
Comment (stevelikesmusic) [inline src/components/TestModal.js:18]: Nice use of `Modal`'s compound components 👍

File: src/components/ListItem.js [hunk 5]
```diff
@@ -55,26 +65,30 @@ const ListItem = ({ item, onDelete, token }) => {
...
+          {purchaseNext}
+        </span>
+        {item.name}
+        <span className="screen-reader-only">
+          Next purchase in {item.next_purchase} days.
+        </span>
+        <button
+          className={isPurchased ? 'purchased' : 'not-purchased'}
+          onClick={onHandle}
+          disabled={isPurchased ? !null : null}
```
Comment (stevelikesmusic) [inline src/components/ListItem.js:78]: Not from this PR, but this expression can be simply `disabled={!isPurchased}`. 
```suggestion
          disabled={!isPurchased}
```

**Review by stevelikesmusic — COMMENTED** [review]
Looks great going with a different approach than the first PR 👏 

It'd be nice if we can still connect to router for the modal. I pushed a branch `modal-to-router` that does just this. Give it a peek if you get stuck syncing the two.

File: src/CSS/TestModal.css (hunk 1 above)
Comment (skillitzimberg) [reply src/CSS/TestModal.css:1]: True, true. I just forgot.

File: src/components/TestModal.js (hunk 4 above)
Comment (laurenmbeatty) [reply src/components/TestModal.js:18]: 🔥

**Review by SaraSweetie — APPROVED** [review]
This works well. It's up to you which you choose. I like either version!

File: src/App.js [hunk 6]
```diff
@@ -22,19 +23,19 @@ function App() {
...
-                </nav>
+                <Switch>
+                  <Route
+                    exact
+                    path="/:itemId?"
+                    render={() => (
+                      <Shopping userToken={userToken} list={data} />
+                    )}
+                  />
+                  <Route
```
Comment (stevelikesmusic) [inline src/App.js:34]: The `/add` route needs to go first @espain16 @skillitzimberg. Switch will match the first path that it matches.

**Pull request closed without merging**